gwt-jackson/src/main/java/com/github/nmorel/gwtjackson/client/utils/DateFormat.java

```diff
@@ -178,13 +178,13 @@ public static Date parse( String date ) {
      *
      * @return the parsed date
      */
-    public static Date parse( String pattern, String date ) {
+    public static Date parse( boolean adjustDatesToContextTimeZone, String pattern, String date ) {
         if ( null == pattern ) {
             return parse( DateFormat.DATE_FORMAT_STR_ISO8601, date );
         } else {
             DateParser parser = CACHE_PARSERS.get( pattern );
             if ( null == parser ) {
-                if ( hasTz( pattern ) ) {
+                if ( hasTz( pattern ) || !adjustDatesToContextTimeZone) {
```

Comment:
Put the condition `!adjustDatesToContextTimeZone` first. Don't need to parse the pattern looking for a timezone if it's true. Will be a bit faster.